pwiz_tools/Skyline/Model/Databinding/Entities/PrecursorResult.cs

```diff
@@ -96,6 +98,68 @@ public void ChangeChromInfo(EditDescription editDescription, Func<TransitionGrou
         public double? LibraryDotProduct { get { return ChromInfo.LibraryDotProduct; } }
         [Format(Formats.STANDARD_RATIO, NullValue = TextUtil.EXCEL_NA)]
         public double? IsotopeDotProduct { get { return ChromInfo.IsotopeDotProduct; } }
+        [Format(Formats.PEAK_AREA)]
+        public double? DeconvolutedPrecursorArea
+        {
+            get
+            {
+                var deconvoluter = new Deconvoluter(SrmDocument.Settings);
+                int? myIndex = null;
+                var myDocNode = Precursor.DocNode;
+                var myUnlabeledAdduct = myDocNode.PrecursorAdduct.Unlabeled;
+                var deconvolutionKeys = new List<DeconvolutionKey>();
+                var areas = new List<KeyValuePair<double, double>>();
+                int replicateIndex = GetResultFile().Replicate.ReplicateIndex;
+                var fileId = GetResultFile().ChromFileInfoId;
+                int optStep = GetResultFile().OptimizationStep;
+
+                foreach (var transitionGroup in Precursor.Peptide.DocNode.TransitionGroups)
+                {
+                    if (!myUnlabeledAdduct.Equals(transitionGroup.PrecursorAdduct.Unlabeled))
+                    {
+                        continue;
+                    }
+                    if (ReferenceEquals(transitionGroup.TransitionGroup, myDocNode.TransitionGroup))
+                    {
+                        myIndex = deconvolutionKeys.Count;
+                    }
+                    deconvolutionKeys.Add(deconvoluter.MakeDeconvolutionKey(Precursor.Peptide.DocNode, transitionGroup));
+                    foreach (var transition in transitionGroup.Transitions)
+                    {
+                        if (!transition.IsMs1)
+                        {
+                            continue;
+                        }
+
+                        if (transition.Results == null ||
+                            transition.Results.Count <= replicateIndex)
+                        {
+                            continue;
+                        }
+
+                        var transitionChromInfo = transition.Results[replicateIndex].FirstOrDefault(
+                            tci => ReferenceEquals(tci?.FileId, fileId) && optStep == tci?.OptimizationStep);
+                        if (transitionChromInfo == null || transitionChromInfo.IsEmpty)
+                        {
+                            continue;
+                        }
+                        areas.Add(new KeyValuePair<double, double>(transition.Mz, transitionChromInfo.Area));
+                    }
+                }
+
+                if (!myIndex.HasValue)
+                {
+                    return null;
+                }
+
+                var minMz = deconvolutionKeys.Min(key => key.TransitionGroupDocNode.PrecursorMz.Value);
+                // TODO(nicksh): decide the window width based on Full Scan settings

```

Comment:
Not much time to do this left. I think we need to do a pass like the VC++ team used to do to expunge all "TODO" comments from the code and make them either "CONSIDER" (if there is an issue that might have meaning but may not be necessary to handle ever), or "FUTURE" if it is something that would be "nice to do some day" (but maybe never).